### PR TITLE
use isset to check if it's set and not null

### PR DESCRIPTION
**Pull request by f0nt4**
Not using isset() will return true if the user_cache exists even though the given uid might not be set.
This can result in an undefined offset error.

### CODE REVIEW

**Comment by Shade-** [comment]
I'd go with `!empty()` rather than `isset()`, as `!empty()` also checks if the variable is not false which is the way this function was designed before. Can you edit the PR @f0nt4? Thank you.

**Comment by f0nt4** [comment]
Using empty() seems to work as well. Changed the PR.

File: inc/functions.php [hunk 1]
```diff
@@ -1193,13 +1193,13 @@ function user_permissions($uid=0)
 	if($uid != $mybb->user['uid'])
 	{
 		// We've already cached permissions for this user, return them.
-		if($user_cache[$uid]['permissions'])
+		if(empty($user_cache[$uid]['permissions']))
```
Comment (Shade-) [inline inc/functions.php]: This must be !empty, otherwise it will always fail.

File: inc/functions.php [hunk 2]
```diff
@@ -1193,13 +1193,13 @@ function user_permissions($uid=0)
...
 		{
 			return $user_cache[$uid]['permissions'];
 		}
 
 		// This user was not already cached, fetch their user information.
-		if(!$user_cache[$uid])
+		if(!empty($user_cache[$uid]))
```
Comment (Shade-) [inline inc/functions.php]: And this should be `empty()`.

**Comment by f0nt4** [comment]
Yup, you're right. Still a little sleepy here,

**Comment by Shade-** [comment]
Not a problem. Thank you for contributing. I shall remind you that, as stated in the [guidelines](https://github.com/mybb/mybb/blob/feature/CONTRIBUTING.md), issues must be opened in the 1.8 bugs and issues community forum, discussed there and approved by a team member.

**Comment by f0nt4** [comment]
Ah ok, didn't know that. Will keep it in mind for the future! Thanks for merging :)

**Pull request merged**